miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java

```diff
@@ -426,6 +426,25 @@ private static DetailedSample toDetailedSample(DetailedSampleDto from) {
     }
     to.setNonStandardAlias(from.getNonStandardAlias());
     to.setParent(getParent(from));
+
+    if (!LimsUtils.isStringEmptyOrNull(from.getExternalNames()) && to.getParent() != null) {
+
+      DetailedSample child = to;
+      DetailedSample parent = to.getParent();
+
+      while (parent.getParent() != null) {
+        child = parent;
+        parent = child.getParent();
+      }
```

Comment:
Can use `LimsUtils.getParent(Identity.class, to)` instead. Will return null if there's no Identity in the hierarchy for some reason, so you'll have to check for null below instead